### PR TITLE
More fixups to the printer post-processing

### PR DESCRIPTION
**Pull request by eric-wieser**
Continues from #378. This PR fixes:
* Zombie equation LHS which can end up being printed multiple times despite being requested once
* Swallowing of `#` and `%` character from the middle of a line
* Messy and inconsistent post-processing whitespace
* The lack of explanation of the post-processor.

It may be instructive to review the  commits individually.

It may be easiest to review with whitespace diffs turned off.

### CODE REVIEW

File: examples/LaTeX/print_check_latex.py [hunk 1]
```diff
@@ -164,6 +164,10 @@ def main():
 
     dirac_eq.Fmt(3,r'\nabla \bm{\psi} I \sigma_{z}-e\bm{A}\bm{\psi}-m\bm{\psi}\gamma_{t} = 0')
 
+    # former bug: "value" ended up on both lines
```
Comment (eric-wieser) [inline examples/LaTeX/print_check_latex.py:167]: See one of the intermediate commits to see the output this used to have

Comment (eric-wieser) [reply examples/LaTeX/print_check_latex.py:167]: https://github.com/pygae/galgebra/pull/379/commits/7b9c5370273eb58964815d24487125466993a6a5

File: galgebra/printer.py [hunk 2]
```diff
@@ -801,10 +801,24 @@ def repl_func(m):
...
+      begins as on. The rest of the line is never processed.
+    * ``#%`` at the start of a line wraps the line in an ``equation`` environment,
+      unless the line already contains an ``align`` environment, in which case
+      everything before the ``align`` environment is moved inside it.
+    * ``#`` at the start of a line disables postprocessing for that line only
+    * Otherwise, lines are split at ``=``. The left-hand-side is post-processed
+      from python syntax into TeX, unless it starts with a ``%``. Finally,
+      the line is wrapped in an ``equation`` environment,
+      unless the line already contains an ``align`` environment, in which case
+      everything before the ``align`` environment is moved inside it
```
Comment (eric-wieser) [inline galgebra/printer.py:820]: Rendering is at least not awful: https://galgebra--379.org.readthedocs.build/en/379/generated/galgebra.printer.html#galgebra.printer.tex

**Comment by eric-wieser** [comment]
LGTM comment was nonsense, caused by the branch switch I did above.

File: galgebra/printer.py [hunk 3]
```diff
@@ -801,10 +801,24 @@ def repl_func(m):
...
 def tex(paper=(14, 11), debug=False, prog=False, pt='10pt'):
-    """
+    r"""
     Post processes LaTeX output (see comments below), adds preamble and
     postscript.
 
+    This postprocessing includes:
+
+    * ``##`` at the start of a line toggles postprocessing on and off. It
+      begins as on. The rest of the line is never processed.
```
Comment (utensil) [inline galgebra/printer.py:811]: "toggle on and off" and "it begins as on" together is confusing.

File: galgebra/printer.py (hunk 2 above)
Comment (utensil) [reply galgebra/printer.py:820]: Oh this logic in code is messy and reflected in the doc......

Comment (eric-wieser) [reply galgebra/printer.py:820]: Absolutely...

File: galgebra/printer.py (hunk 3 above)
Comment (eric-wieser) [inline galgebra/printer.py:811]:
```suggestion
    * ``##`` at the start of a line toggles between enabling and disabling
      postprocessing for subsequent lines. The text after this marker is never
      postprocessed.
```

Comment (eric-wieser) [reply galgebra/printer.py:811]: Better as https://github.com/pygae/galgebra/pull/379#discussion_r429364677?

File: galgebra/printer.py (hunk 2 above)
Comment (utensil) [reply galgebra/printer.py:820]: I would suggest the following structure of the doc:

- Start by stating that the postprocess includes two parts:
  1. texifying lhs
  2. wrapping it with equation env unless already wrapped with align
- Then explain the various markers:
  - # disables postprocessing for that line only
  - % disables the part 1 of postprocessing for that line only
  - #% (seems to be exactly the same as only %?)
  - ## toggles postprocessing, i.e. the first ## turns it off, the second ## turns it back on, and so on (The text after this marker is never postprocessed.)

(Sorry I didn't spell every word out, only sketching the big picture.

File: galgebra/printer.py [hunk 4]
```diff
@@ -825,47 +839,39 @@ def tex(paper=(14, 11), debug=False, prog=False, pt='10pt'):
...
-            else:  # normal sympy equation
-                latex_line = latex_line.strip()
-                if len(latex_line) > 0:
-                    latex_line = '\\begin{equation*} ' + latex_line + ' \\end{equation*}'
+                latex_line = lhs + '= ' + latex_line.lstrip()
+
+            if r'\begin{align*}' in latex_line:
+                latex_line = r'\begin{align*} ' + latex_line.replace(r'\begin{align*}', '').lstrip()
+            else:
+                latex_line = '\\begin{equation*} ' + latex_line.strip() + ' \\end{equation*}'
```
Comment (utensil) [inline galgebra/printer.py:874]: Should use r-string consistently

Comment (eric-wieser) [reply galgebra/printer.py:874]: Good catch

File: galgebra/printer.py (hunk 2 above)
Comment (eric-wieser) [reply galgebra/printer.py:820]: You're right, `%` and `#%` look to be practically identical...

Comment (eric-wieser) [reply galgebra/printer.py]: Updated to merge their implementations, the only effect was insignificant whitespace.

**Pull request merged**